### PR TITLE
Add scripts for building an iso

### PR DESCRIPTION
**Pull request by dlorenc**
This script allows us to build and release an iso with dependencies inside it. The iso is based on boot2docker, and the build scripts were heavily inspired by https://github.com/skippbox/boot2k8s

We don't bake in any kubernetes tooling yet, the only thing so far is installing nsenter and pinning docker to a particular version.

### CODE REVIEW

**Comment by dlorenc** [comment]
@runseb Thanks for the pointer to boot2k8s earlier. Mind taking a quick look here and making sure it looks reasonable?

One difference I noticed is that I'm grabbing nsenter from outside of the build, where you appear to be doing it at boot time.

**Comment by ethernetdan** [comment]
Do we want to verify the integrity+identity of the Docker tarball?

Also should we distribute a signature with this ISO?

**Comment by dlorenc** [comment]
> Do we want to verify the integrity+identity of the Docker tarball?
> 
Good idea. Added a check in the Dockerfile.
> Also should we distribute a signature with this ISO?
Yeah, we definitely should. I'd like to bundle these all together at some point though, and only use the separate downloads for development.

Thanks for looking.

File: iso/build.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# Output the iso.
+docker run iso > $ISO
+
+popd
+mv $tmpdir/$ISO .
+
+# Clean up.
+rm -rf $tmpdir
+
+echo "Iso available at ./$ISO"
```
Comment (luxas) [inline iso/build.sh]: nit: newline

File: iso/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+minikube.iso
```
Comment (luxas) [inline iso/.gitignore]: nit: newline

Comment (dlorenc) [inline iso/.gitignore]: Done.

File: iso/build.sh (hunk 1 above)
Comment (dlorenc) [inline iso/build.sh]: Done.

**Pull request merged**